tdp/cli/commands/plan/edit.py

```diff
@@ -0,0 +1,214 @@
+
+    Args:
+        line: line to be parsed
+
+    Returns:
+        Operation, host and extra vars
+    """
+    parsed_line = re.match(
+        r"^(.*?)( on .*?){0,1}( ?with .*?){0,1}( ?on .*?){0,1}$", line
+    )
+
+    if parsed_line is None:
+        raise ValueError(
+            "Error on line '"
+            + line
+            + "' : it must be 'OPERATION [on HOST] [with EXTRA_VARS[,EXTRA_VARS]].'"
+        )
+
+    if parsed_line.group(1).split(" ")[0] == "":
+        raise ValueError("Error on line '" + line + "' : it must contain an operation.")
+
+    if len(parsed_line.group(1).strip().split(" ")) > 1:
+        raise ValueError("Error on line '" + line + "' : only 1 operation is allowed.")
+
+    if parsed_line.group(2) is not None and parsed_line.group(4) is not None:
+        raise ValueError(
+            "Error on line '" + line + "' : only 1 host is allowed in a line."
+        )
+
+    operation = parsed_line.group(1).strip()
+
+    # Get the host and test if it is declared
+    if parsed_line.group(2) is not None:
+        host = parsed_line.group(2).split(" on ")[1]
+        if host == "":
+            raise ValueError(
+                "Error on line '" + line + "' : host is required after 'on' keyword."
+            )
+    elif parsed_line.group(4) is not None:
+        host = parsed_line.group(4).split(" on ")[1]
+        if host == "":
+            raise ValueError(
+                "Error on line '" + line + "' : host is required after 'on' keyword."
+            )
+    else:
+        host = None
+
+    # Get the extra vars and test if they are declared
+    if parsed_line.group(3) is not None:
+        extra_vars = parsed_line.group(3).split(" with ")[1]
+        if extra_vars == "":
+            raise ValueError("Extra vars are required after 'with' keyword.")
+        extra_vars = extra_vars.split(",")
+        extra_vars = [item.strip() for item in extra_vars]
+    else:
+        extra_vars = None
+
+    return (operation, host, extra_vars)
+
+
+@click.command(short_help="Edit Deployment plan.")
+@collections
+@database_dsn
+def edit(
+    collections,
+    database_dsn,
+):
+    with get_session(database_dsn, commit_on_exit=True) as session:
+        planned_deployment_log = get_planned_deployment_log(session)
+        if planned_deployment_log is None:
+            raise click.ClickException(
+                "No planned deployment found, please run `tdp plan` first."
+            )
+
+        operation_list = [
+            (operation.operation, operation.host, operation.extra_vars)
+            for operation in planned_deployment_log.operations
+        ]
+
+        operation_lines = [
+            operation
+            + (" on " + host if host else "")
+            + (" with " + ",".join(extra_vars) if extra_vars else "")
+            for operation, host, extra_vars in operation_list
+        ]
+
+        temp_file = tempfile.NamedTemporaryFile(mode="w", suffix=".csv", delete=False)
```

Comment:
It seems that you can use [`tempfile.NamedTemporaryFile`](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryFile) as a context manager. It feels cleaner as they provide automatic cleanup.

Reply:
Done, I used a context manager to handle the temporary file, thx for the help ^^